### PR TITLE
Fixing Build errors in remove_extra_typedef

### PR DESCRIPTION
**Pull request by yashpatel007**
- [x] Does this PR have a descriptive title that could go in our release notes?
- [x] Does this PR add any new dependencies?
- [x] Does this PR modify any existing APIs?
   - [x] Is the change to the API backwards compatible?
- [x] Should this result in any changes to our documentation, either updating existing docs or adding new ones?

### CODE REVIEW

File: include/utils.h [hunk 1]
```diff
@@ -504,8 +494,8 @@ inline void prune_truthset_for_range(const std::string &bin_file, float range,
     reader.read((char *)&npts_i32, sizeof(int));
```
Comment (gopalrs) [inline include/utils.h:494]: It would also be nice to use the get_metadata (and similar functions) to avoid repeated code

File: include/utils.h [hunk 2]
```diff
@@ -558,23 +548,24 @@ inline void prune_truthset_for_range(const std::string &bin_file, float range,
...
-    _u64 read_blk_size = 64 * 1024 * 1024;
+    uint64_t read_blk_size = 64 * 1024 * 1024;
     cached_ifstream reader(bin_file, read_blk_size);
     diskann::cout << "Reading truthset file " << bin_file.c_str() << "... " << std::flush;
     size_t actual_file_size = reader.get_file_size();
 
-    int npts_u32, total_u32;
-    reader.read((char *)&npts_u32, sizeof(int));
-    reader.read((char *)&total_u32, sizeof(int));
+    int nptsuint32_t, totaluint32_t;
```
Comment (gopalrs) [inline include/utils.h:559]: Use get_metadata()?

File: include/utils.h
Comment (gopalrs) [inline include/utils.h:1]: I think in the next round of refactoring we should split utils.h into smaller files with more cohesive functionality.

File: include/utils.h [hunk 3]
```diff
@@ -813,17 +804,17 @@ template <typename T> float prepare_base_for_inner_products(const std::string in
     std::cout << "Pre-processing base file by adding extra coordinate" << std::endl;
     std::ifstream in_reader(in_file.c_str(), std::ios::binary);
     std::ofstream out_writer(out_file.c_str(), std::ios::binary);
-    _u64 npts, in_dims, out_dims;
+    uint64_t npts, in_dims, out_dims;
     float max_norm = 0;
 
-    _u32 npts32, dims32;
+    uint32_t npts32, dims32;
```
Comment (gopalrs) [inline include/utils.h:810]: Use get_metadata()

File: src/partition.cpp [hunk 4]
```diff
@@ -409,29 +409,29 @@ int shard_data_into_clusters_only_ids(const std::string data_file, float *pivots
...
 {
-    _u64 read_blk_size = 64 * 1024 * 1024;
-    //  _u64 write_blk_size = 64 * 1024 * 1024;
+    uint64_t read_blk_size = 64 * 1024 * 1024;
+    //  uint64_t write_blk_size = 64 * 1024 * 1024;
     // create cached reader + writer
     cached_ifstream base_reader(data_file, read_blk_size);
-    _u32 npts32;
-    _u32 basedim32;
+    uint32_t npts32;
```
Comment (gopalrs) [inline src/partition.cpp:416]: Use get_metadata()?

File: src/pq_flash_index.cpp [hunk 5]
```diff
@@ -13,27 +13,28 @@
 #include "linux_aligned_file_reader.h"
 #endif
 
-#define READ_U64(stream, val) stream.read((char *)&val, sizeof(_u64))
-#define READ_U32(stream, val) stream.read((char *)&val, sizeof(_u32))
+#define READ_U64(stream, val) stream.read((char *)&val, sizeof(uint64_t))
```
Comment (gopalrs) [inline src/pq_flash_index.cpp:16]: Can we move these into inline functions?

File: src/utils.cpp [hunk 6]
```diff
@@ -100,24 +100,25 @@ void normalize_data_file(const std::string &inFileName, const std::string &outFi
     std::ofstream writr(outFileName, std::ios::binary);
 
     int npts_s32, ndims_s32;
-    readr.read((char *)&npts_s32, sizeof(_s32));
-    readr.read((char *)&ndims_s32, sizeof(_s32));
+    readr.read((char *)&npts_s32, sizeof(int32_t));
```
Comment (gopalrs) [inline src/utils.cpp:103]: get_metadata() here as well?

File: tests/utils/bin_to_fvecs.cpp [hunk 7]
```diff
@@ -25,31 +26,31 @@ int main(int argc, char **argv)
     std::ifstream readr(argv[1], std::ios::binary);
     int npts_s32;
     int ndims_s32;
-    readr.read((char *)&npts_s32, sizeof(_s32));
-    readr.read((char *)&ndims_s32, sizeof(_s32));
+    readr.read((char *)&npts_s32, sizeof(int32_t));
```
Comment (gopalrs) [inline tests/utils/bin_to_fvecs.cpp:29]: get_metadata()?

File: tests/utils/bin_to_tsv.cpp [hunk 8]
```diff
@@ -36,22 +37,22 @@ int main(int argc, char **argv)
     }
 
     std::ifstream reader(argv[2], std::ios::binary);
-    _u32 npts_u32;
-    _u32 ndims_u32;
-    reader.read((char *)&npts_u32, sizeof(_s32));
-    reader.read((char *)&ndims_u32, sizeof(_s32));
+    uint32_t npts_u32;
```
Comment (gopalrs) [inline tests/utils/bin_to_tsv.cpp:40]: get_metadata()?

File: tests/utils/float_bin_to_int8.cpp [hunk 9]
```diff
@@ -28,29 +28,29 @@ int main(int argc, char **argv)
     }
 
     std::ifstream reader(argv[1], std::ios::binary);
-    _u32 npts_u32;
-    _u32 ndims_u32;
-    reader.read((char *)&npts_u32, sizeof(_s32));
-    reader.read((char *)&ndims_u32, sizeof(_s32));
+    uint32_t npts_u32;
```
Comment (gopalrs) [inline tests/utils/float_bin_to_int8.cpp:31]: get_metadata()?

File: tests/utils/int8_to_float_scale.cpp [hunk 10]
```diff
@@ -28,29 +28,29 @@ int main(int argc, char **argv)
     }
 
     std::ifstream reader(argv[1], std::ios::binary);
-    _u32 npts_u32;
-    _u32 ndims_u32;
-    reader.read((char *)&npts_u32, sizeof(_s32));
-    reader.read((char *)&ndims_u32, sizeof(_s32));
+    uint32_t npts_u32;
```
Comment (gopalrs) [inline tests/utils/int8_to_float_scale.cpp:31]: get_metadata()?

**Review by gopalrs — CHANGES_REQUESTED** [review]
- Can we also change all unsigned's to uint32_ts?

File: include/scratch.h [hunk 11]
```diff
@@ -22,7 +22,7 @@
 // SSD Index related limits
 #define MAX_GRAPH_DEGREE 512
```
[the remark this answers is not in the document]
Comment (harsha-simhadri) [reply include/scratch.h:23]: Lets move constants to a single file in another PE.

File: tests/utils/int8_to_float_scale.cpp (hunk 10 above)
Comment (harsha-simhadri) [reply tests/utils/int8_to_float_scale.cpp:31]: Lets take this up separately https://github.com/microsoft/DiskANN/issues/294

File: include/utils.h [hunk 12]
```diff
@@ -53,7 +51,7 @@ typedef int FileHandle;
     4096 // all metadata of individual sub-component files is written in first
          // 4KB for unified files
 
-#define BUFFER_SIZE_FOR_CACHED_IO (_u64)1024 * (_u64)1048576
+#define BUFFER_SIZE_FOR_CACHED_IO (uint64_t)1024 * (uint64_t)1048576
```
Comment (harsha-simhadri) [inline include/utils.h:54]: could be size_t?

File: src/partition.cpp [hunk 13]
```diff
@@ -33,7 +33,7 @@
 template <typename T>
 void gen_random_slice(const std::string base_file, const std::string output_prefix, double sampling_rate)
 {
-    _u64 read_blk_size = 64 * 1024 * 1024;
+    uint64_t read_blk_size = 64 * 1024 * 1024;
```
Comment (harsha-simhadri) [inline src/partition.cpp:36]: size_t?

**Review by harsha-simhadri — COMMENTED** [review]
use size_t for _u64 where it is intended for array offset calculation

**Review by harsha-simhadri — APPROVED** [review]

File: include/pq.h [hunk 14]
```diff
@@ -17,10 +17,10 @@ namespace diskann
 class FixedChunkPQTable
 {
     float *tables = nullptr; // pq_tables = float array of size [256 * ndims]
-    _u64 ndims = 0;          // ndims = true dimension of vectors
-    _u64 n_chunks = 0;
```
Comment (rakri) [inline include/pq.h:21]: should be uint64_t as they are not sizes

File: include/pq.h [hunk 15]
```diff
@@ -86,15 +87,16 @@ template <typename T> struct PQScratch
     }
 };
 
-void aggregate_coords(const std::vector<unsigned> &ids, const _u8 *all_coords, const _u64 ndims, _u8 *out);
```
Comment (rakri) [inline include/pq.h:89]: same for ndims, shd be uint64_t

File: include/pq.h [hunk 16]
```diff
@@ -86,15 +87,16 @@ template <typename T> struct PQScratch
...
+void aggregate_coords(const std::vector<unsigned> &ids, const uint8_t *all_coords, const size_t ndims, uint8_t *out);
 
-void pq_dist_lookup(const _u8 *pq_ids, const _u64 n_pts, const _u64 pq_nchunks, const float *pq_dists,
```
Comment (rakri) [inline include/pq.h:91]: same for pq_nchunks

File: include/pq.h [hunk 17]
```diff
@@ -86,15 +87,16 @@ template <typename T> struct PQScratch
...
+void pq_dist_lookup(const uint8_t *pq_ids, const size_t n_pts, const size_t pq_nchunks, const float *pq_dists,
                     std::vector<float> &dists_out);
 
 // Need to replace calls to these with calls to vector& based functions above
-void aggregate_coords(const unsigned *ids, const _u64 n_ids, const _u8 *all_coords, const _u64 ndims, _u8 *out);
```
Comment (rakri) [inline include/pq.h:95]: n_ids

Comment (rakri) [reply include/pq.h:95]: n_dims

File: include/pq_flash_index.h [hunk 18]
```diff
@@ -123,25 +125,25 @@ template <typename T, typename LabelT = uint32_t> class PQFlashIndex
...
     // data info
-    _u64 num_points = 0;
-    _u64 num_frozen_points = 0;
-    _u64 frozen_location = 0;
-    _u64 data_dim = 0;
-    _u64 disk_data_dim = 0; // will be different from data_dim only if we use
-                            // PQ for disk data (very large dimensionality)
-    _u64 aligned_dim = 0;
-    _u64 disk_bytes_per_point = 0;
+    size_t num_points = 0;
```
Comment (rakri) [inline include/pq_flash_index.h:128]: these should also be uint64_t? i am thinking only file sizes, memory, offsets etc shd be size_t?

File: include/utils.h [hunk 19]
```diff
@@ -834,19 +825,19 @@ template <typename T> float prepare_base_for_inner_products(const std::string in
...
 
     std::vector<float> norms(npts, 0);
 
-    for (_u64 b = 0; b < num_blocks; b++)
+    for (uint64_t b = 0; b < num_blocks; b++)
     {
-        _u64 start_id = b * block_size;
-        _u64 end_id = (b + 1) * block_size < npts ? (b + 1) * block_size : npts;
-        _u64 block_pts = end_id - start_id;
+        size_t start_id = b * block_size;
```
Comment (rakri) [inline include/utils.h:834]: again size_t vs uint64?

File: src/disk_utils.cpp [hunk 20]
```diff
@@ -239,43 +239,43 @@ void read_idmap(const std::string &fname, std::vector<unsigned> &ivecs)
...
+                 const std::string &idmaps_suffix, const uint64_t nshards, uint32_t max_degree,
                  const std::string &output_vamana, const std::string &medoids_file, bool use_filters,
                  const std::string &labels_to_medoids_file)
 {
     // Read ID maps
     std::vector<std::string> vamana_names(nshards);
-    std::vector<std::vector<unsigned>> idmaps(nshards);
-    for (_u64 shard = 0; shard < nshards; shard++)
+    std::vector<std::vector<uint32_t>> idmaps(nshards);
+    for (size_t shard = 0; shard < nshards; shard++)
```
Comment (rakri) [inline src/disk_utils.cpp:249]: uint64 vs size_t

File: src/index.cpp [hunk 21]
```diff
@@ -1316,14 +1322,14 @@ void Index<T, TagT, LabelT>::inter_insert(unsigned n, std::vector<unsigned> &pru
...
-        std::vector<unsigned> copy_of_neighbors;
+        std::vector<uint32_t> copy_of_neighbors;
         bool prune_needed = false;
         {
             LockGuard guard(_locks[des]);
             auto &des_pool = _final_graph[des];
             if (std::find(des_pool.begin(), des_pool.end(), n) == des_pool.end())
             {
-                if (des_pool.size() < (_u64)(GRAPH_SLACK_FACTOR * range))
+                if (des_pool.size() < (size_t)(GRAPH_SLACK_FACTOR * range))
```
Comment (rakri) [inline src/index.cpp:1332]: size_t vs uint64

File: src/pq.cpp [hunk 22]
```diff
@@ -41,16 +41,16 @@ void FixedChunkPQTable::load_pq_centroid_bin(const char *pq_table_file, size_t n
 {
 #endif
 
-    _u64 nr, nc;
+    size_t nr, nc;
```
Comment (rakri) [inline src/pq.cpp:44]: size_t vs uint64

**Review by rakri — COMMENTED** [review]
looks good, except for uint64_t vs size_t

**Review by gopalrs — APPROVED** [review]

**Review by rakri — APPROVED** [review]

**Pull request merged**